src/harness/sas_test_harness.py

```diff
@@ -0,0 +1,360 @@
+    """ This constructor initializes the SasTestHarnessServer class with required information
+    passed as args.
+
+    Args:
+        name: A name for SAS Test Harness.If multiple SAS Test Hareness are created
+              then this value should be unique.
+        host_name: host name of SAS Test Harness that the SAS UUT should connect to for FAD records.
+        port:   port of SAS Test Harness host where http server is configured.
+        sas_version: sas version of the SAS Test.
+        dump_path: absolute file_path where the FAD records served by
+              the SAS Test Harness will be placed.
+        cert_file: The absolute path to the certificate used by SAS Test Harness
+              to authorize with SAS UUT.
+        key_file:  The absolute path to the private key file for the cert_file.
+    """
+    super(SasTestHarnessServer, self).__init__()
+    self.name = name
+    self.host_name = host_name
+    self.port = port
+    self.sas_version = sas_version
+
+    #BaseURL format should be https: // < hostname >: < port > / versionX.Y
+    self.http_server_url = "https://" + host_name + ':' + str(port) + '/' + sas_version
+    self.dump_path = dump_path
+    self.cert_file = cert_file if cert_file is not None else CERT_FILE
+    self.key_file = key_file if key_file is not None else KEY_FILE
+    self.server = None
+    self.setDaemon(True)
+
+  def __del__(self):
+     SasTestHarnessServer.cleanDumpFiles(self.dump_path)
+
+  def getBaseUrl(self):
+    return self.http_server_url
+
+  def getUrlHostNameAndPort(self):
+    """ This method will return the http_server_url used in startServer method by truncating other
+    querystring in url and returns tuple(<hostname>, <port>).
+    """
+    return (self.host_name, self.port)
+
+  def run(self):
+    """ Starts the HTTPServer as background thread.
+    """
+    self.server = SasHttpServer(self.dump_path,
+                                self.getUrlHostNameAndPort(),
+                                SasTestHarnessServerHandler)
+    self.server.socket = ssl.wrap_socket(
+        self.server.socket,
+        certfile=self.cert_file,
+        keyfile=self.key_file,
+        ca_certs=CA_CERT,
+        cert_reqs=ssl.CERT_REQUIRED,
+        ssl_version=ssl.PROTOCOL_TLSv1_2,
+        ciphers=':'.join(CIPHERS),
+        server_side=True
+    )
+
+    # Start HTTP Server.
+    logging.info("Started Test Harness Server:%s at %s", self.name, self.http_server_url)
+    self.server.serve_forever()
+
+  def shutdown(self):
+    self.server.shutdown()
+    logging.info("Stopped Test Harness Server:%s", self.name)
+
+  @staticmethod
+  def getCbsdReferenceId(fcc_id, serial_number):
+    """creates encoded cbsd_reference_id using sha1 with fcc_id and serial number of cbsd.
+    """
+    return str(fcc_id + '/' + str(hashlib.sha1(serial_number).hexdigest())).encode('utf-8')
+
+  @staticmethod
+  def writeDumpFile(base_path, file_name, data):
```

Comment:
This   was made a static method to allow record creation without creating a   SasTestHarness object.           This has been changed. Now it has been made as private method so that it   uses the temp directory of the dump path.